core/src/Cuda/Kokkos_Cuda_KernelLaunch.hpp

```diff
@@ -160,13 +161,16 @@ inline void configure_shmem_preference(KernelFuncPtr const& func,
                                        bool prefer_shmem) {
 #ifndef KOKKOS_ARCH_KEPLER
   // On Kepler the L1 has no benefit since it doesn't cache reads
-  static bool cache_config_preference_cached = [&] {
+  auto set_cache_config = [&] {
```

Comment:
```suggestion
  static auto set_cache_config = [&] {
```

(I'm not sure if `constexpr` can also be added here)